### PR TITLE
Bael 2330

### PR DESCRIPTION
**Pull request by kyleandari**
Adding removeDuplicatesUsingDistinct
Adding more asserts to the unit tests

### CODE REVIEW

File: java-strings/src/test/java/com/baeldung/stringduplicates/RemoveDuplicateFromStringUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public void executedBeforeEach() {
+        removeDuplicateFromString = new RemoveDuplicateFromString();
+    }
+
+
+    @Test
+    public void whenUsingCharArray_DuplicatesShouldBeRemovedWithoutKeepingStringOrder() {
+        String str1 = removeDuplicateFromString.removeDuplicatesUsingCharArray(STR1);
+        String str2 = removeDuplicateFromString.removeDuplicatesUsingCharArray(STR2);
+        Assert.assertEquals("", "");
```
Comment (echeyne) [inline java-strings/src/test/java/com/baeldung/stringduplicates/RemoveDuplicateFromStringUnitTest.java]: What is the point of these empty string asserts?

File: java-strings/src/main/java/com/baeldung/stringduplicates/RemoveDuplicateFromString.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        for (Character c : linkedHashSet) {
+            sb.append(c);
+        }
+
+        return sb.toString();
+    }
+
+    String removeDuplicatesUsingSorting(String str) {
```
Comment (echeyne) [inline java-strings/src/main/java/com/baeldung/stringduplicates/RemoveDuplicateFromString.java]: As @jzheaux mentioned this method will still throw an exception if passed an empty string. Can you fix this up?

**Comment by echeyne** [comment]
Can you resolve your merge conflicts by merging in the master tutorials branch?

File: java-strings/src/test/java/com/baeldung/stringduplicates/RemoveDuplicateFromStringUnitTest.java (hunk 1 above)
Comment (kyleandari) [reply java-strings/src/test/java/com/baeldung/stringduplicates/RemoveDuplicateFromStringUnitTest.java]: I was trying to test for empty strings but I forget to add the below line
 String strEmpty = removeDuplicateFromString.removeDuplicatesUsingSorting(STR_EMPTY);
will be fixed

File: java-strings/src/main/java/com/baeldung/stringduplicates/RemoveDuplicateFromString.java (hunk 2 above)
Comment (kyleandari) [reply java-strings/src/main/java/com/baeldung/stringduplicates/RemoveDuplicateFromString.java]: will add an if statement

File: java-strings/src/test/java/com/baeldung/stringduplicates/RemoveDuplicateFromStringUnitTest.java (hunk 1 above)
Comment (kyleandari) [reply java-strings/src/test/java/com/baeldung/stringduplicates/RemoveDuplicateFromStringUnitTest.java]: Done

File: java-strings/src/main/java/com/baeldung/stringduplicates/RemoveDuplicateFromString.java (hunk 2 above)
Comment (kyleandari) [reply java-strings/src/main/java/com/baeldung/stringduplicates/RemoveDuplicateFromString.java]: Done

**Pull request merged**